src/mpyl/steps/deploy/k8s/chart.py

```diff
@@ -243,16 +243,49 @@ def _to_k8s_model(values: dict, model_type):
         )
 
     @staticmethod
-    def _to_probe(probe: Probe, defaults: dict, target: Target) -> V1Probe:
+    def _to_probe(probe: Optional[Probe], defaults: dict, target: Target) -> V1Probe:
         values = defaults.copy()
-        values.update(probe.values)
+        if probe:
+            values.update(probe.values)
         v1_probe: V1Probe = ChartBuilder._to_k8s_model(values, V1Probe)
-        path = probe.path.get_value(target)
+        path = probe.path.get_value(target) if probe else None
         v1_probe.http_get = V1HTTPGetAction(
             path="/health" if path is None else path, port="port-0"
         )
         return v1_probe
 
+    def _construct_probes(self) -> tuple[Optional[V1Probe], Optional[V1Probe]]:
+        """
+        Construct kubernetes probes based on project yaml values and default values in mpyl_config.yaml.
+        Add a startup probe only if liveness_probe is defined
+
+        NOTE: in MPL the same liveness probe was copied for startup probe,
+              if no startup probe was provided in the project yaml.
+              here we're using the startup probe default values from the config instead
```

Comment:
I think the MPL reference is not going to be understood soon. Perhaps you could just stick to describing the current (fallback to default values) mechanism?

Reply:
fair enough!